pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaMPI.hpp

```diff
@@ -262,11 +262,9 @@ class StateVectorCudaMPI final
                        const std::size_t index, const bool async = false) {
         std::size_t rankId = index >> BaseType::getNumQubits();
 
+        constexpr std::size_t one{1U};
         std::size_t local_index =
-            static_cast<std::size_t>(
-                rankId * std::pow(2.0, static_cast<long double>(
-                                           BaseType::getNumQubits()))) ^
-            index;
+            rankId * (one << BaseType::getNumQubits()) ^ index;
```

Comment:
` const std::size_t local_index = rankId * (one << BaseType::getNumQubits()) ^ index;
`
Should we mark it as const?

Reply:
Do we want to do bound check for the left shift operation here? It would not be a problem for now but be good for long run.

Reply:
> You are right. I think we can add a `TODO` note for the moment. What do you think? :thinking: